long-read-mngs/run.wdl

```diff
@@ -455,18 +455,59 @@ task TallyHits {
             --m8-filepath "~{m8}" \
             --hitsummary-filepath "~{hitsummary}" \
             --reads-to-contigs-filepath reads_to_contigs.txt \
-            --output-filepath "tallied_hits.csv" \
+            --output-filepath "tallied_hits.csv" 
     >>>
 
     output{
         File tallied_hits = "tallied_hits.csv"
+        File reads_to_contigs = "reads_to_contigs.txt"
     }
 
     runtime {
         docker: docker_image_id
     }
 }
 
+task UnmappedReads {
+    input {
+        File input_file
+        File hitsummary_nt
+        File hitsummary_nr
+        File reads_to_contigs_txt
+        String docker_image_id
+    }
+    command <<<
+        set -euxo pipefail
+        cut -f1 "~{hitsummary_nt}" "~{hitsummary_nr}" | sort | uniq > mapped.txt
+        cut -f1,2 "~{reads_to_contigs_txt}" > all_reads.txt
+        python3 << CODE
+        with open("mapped.txt", "r") as m:
+            mapped_hits = set(m.read().splitlines())
+
+        unmapped_reads = set()
+        with open("all_reads.txt", "r") as ar:
+            for line in ar:
+                read, contig = line.strip().split("\t")
+                if read not in mapped_hits and contig not in mapped_hits:
+                    unmapped_reads.add(read)
+
+        with open("unmapped_reads.txt", "w") as ur:
+            for read in unmapped_reads:
```

Comment:
I think this can just be:

```python
ur.writelines(unmapped_reads)
```

Reply:
I think it doesn't output with the newlines, possibly because it's a set?